projects/core/src/features-config/feature-toggles/config/feature-toggles.ts

```diff
@@ -724,6 +724,19 @@ export interface FeatureTogglesInterface {
    * Enabling this flag removes the margin animation to improve performance and user experience.
    */
   disableCxPageSlotMarginAnimation?: boolean;
+
+  /**
+   * Feature flag to enable using <link rel=preconnect> in the index.html.
+   *
+   * ## Why this flag exists
+   * 1. **Preconnecting is only effective for domains other than the origin domain, so you shouldn't use it for your site.
```

Comment:
Good observation in the JSDoc!

It inspired me to this question:
What about adding OOTB the following logic: let's check in `MediaPreconnectService` if the domain current storefront's domain rendered in the SSR (`windowRef.location.origin`) is the same as the _domain_ of the configured media base url? (note: mediaBaseUrl may contain some _/url/path_ besides the _domain_ too).

And if the domains the same, then the preconnect tag shouldn't be added in SSR (regardless of the toggle).

Then in the toggle's JSDoc we could mention:
```suggestion
* 1. **Preconnecting is not needed (and won't be performed) if the domain of the media base url is the same as the storefront's domain.
```